### PR TITLE
Review feedback

### PR DESCRIPTION
**Pull request by mcordova47**
* Better documentation with examples
* Inline helpers

### CODE REVIEW

File: src/Elmish/Hooks/UseEffect.purs [hunk 1]
```diff
@@ -5,19 +5,30 @@ module Elmish.Hooks.UseEffect
...
-useEffect name init =
-  mkHook name \n render -> useEffect' n { init, render }
+useEffect name initAff =
+  mkHook name \n render -> useEffect' n { init: initAff, render }
+  where
+    useEffect' n = wrapWithLocalState n \{ init, render } ->
+      { init: forkVoid init
+      , update: \_ msg -> absurd msg
+      , view: \_ _ -> render unit
+      }
```
Comment (fsoikin) [inline src/Elmish/Hooks/UseEffect.purs:34]: It doesn't really need to pass `initAff` as a parameter, does it? And once you get rid of that, the only remaining parameter is `render`, and now you can eta-reduce:

```suggestion
  mkHook name useEffect'
  where
    useEffect' n = wrapWithLocalState n \render ->
      { init: forkVoid initAff
      , update: \_ msg -> absurd msg
      , view: \_ _ -> render unit
      }
```

But better yet, `useEffect'` could just be inline:

```suggestion
  mkHook name \n -> wrapWithLocalState n \render ->
      { init: forkVoid initAff
      , update: \_ msg -> absurd msg
      , view: \_ _ -> render unit
      }
```

File: src/Elmish/Hooks/UseState.purs [hunk 2]
```diff
@@ -7,24 +7,32 @@ import Prelude
...
+-- | ```
 useState :: forall state. HookName -> state -> Hook (state /\ Dispatch state)
 useState name initialState =
   mkHook name \n render -> useState' n { initialState, render }
+  where
+    useState' n = wrapWithLocalState n \{ initialState: init, render } ->
+      { init: pure init
+      , update: \_ newState -> pure newState
+      , view: curry render
+      }
```
Comment (fsoikin) [inline src/Elmish/Hooks/UseState.purs:37]: Same here

File: src/Elmish/Hooks/UseEffect.purs (hunk 1 above)
Comment (mcordova47) [reply src/Elmish/Hooks/UseEffect.purs:34]: Oh right!

Comment (mcordova47) [reply src/Elmish/Hooks/UseEffect.purs]: Well after seeing this, I got to thinking. It doesn’t make much sense to pretend that `mkHook` is more general than it is. The callback currently accepts a `ComponentName`, which really will only get passed to `wrapWithLocalState`. If there’s ever a use case for using some other method of creating an element, we can think about refactoring then. But for now, this is a much simpler API.

Comment (fsoikin) [reply src/Elmish/Hooks/UseEffect.purs]: Yeah, makes sense to me

**Pull request merged**